File: AZ-204_Develop_Solutions_That_Use_Azure_Cosmos_DB/code/resource_mgmnt.py
from azure.cosmosdb.sql import CosmosDBClient
from azure.identity import DefaultAzureCredential

# Set the Azure subscription ID and resource group name
subscription_id = "<your-subscription-id>"
resource_group_name = "<your-resource-group-name>"

# Set the Cosmos DB account name and database name
account_name = "<your-cosmosdb-account-name>"
database_name = "<your-database-name>"

# Create a Cosmos DB client
credential = DefaultAzureCredential()
client = CosmosDBClient(credential, subscription_id)

# Provision a Cosmos DB account
def provision_cosmosdb():
    client.cosmosdb_accounts.begin_create_or_update(
        resource_group_name,
        account_name,
        {
            "location": "<your-location>",
            "kind": "GlobalDocumentDB",
            "database_account_offer_type": "Standard",
            "capabilities": [
                {
                    "name": "EnableServerless"
                }
            ]
        }
    ).result()
    print("Cosmos DB account provisioned successfully.")

# List all Cosmos DB accounts in the resource group
def list_cosmosdb_accounts():
    accounts = client.cosmosdb_accounts.list_by_resource_group(resource_group_name)
    for account in accounts:
        print(f"Account name: {account.name}, Location: {account.location}")

# Update the Cosmos DB account
def update_cosmosdb_account():
    account = client.cosmosdb_accounts.get(resource_group_name, account_name)
    account.capabilities.append({"name": "EnableAutomaticFailover"})
    client.cosmosdb_accounts.begin_create_or_update(resource_group_name, account_name, account).result()
    print("Cosmos DB account updated successfully.")

# Delete the Cosmos DB account
def delete_cosmosdb_account():
    client.cosmosdb_accounts.begin_delete(resource_group_name, account_name).result()
    print("Cosmos DB account deleted successfully.")

# Uncomment the function calls below to execute the operations

# provision_cosmosdb()
# list_cosmosdb_accounts()
# update_cosmosdb_account()
# delete_cosmosdb_account()

File: AZ-204_Develop_Solutions_That_Use_Azure_Cosmos_DB/code/resource_mgmnt.cs
using System;
using System.Threading.Tasks;
using Azure;
using Azure.Identity;
using Azure.ResourceManager.CosmosDB;
using Azure.ResourceManager.CosmosDB.Models;

class Program
{
    static async Task Main(string[] args)
    {
        string subscriptionId = "<your-subscription-id>";
        string resourceGroupName = "<your-resource-group-name>";
        string accountName = "<your-cosmosdb-account-name>";

        // Authenticate using Azure CLI credentials
        var credential = new DefaultAzureCredential();

        // Create a Cosmos DB management client
        var client = new CosmosDBManagementClient(subscriptionId, credential);

        // Provision a new Cosmos DB account
        var account = new DatabaseAccountCreateUpdateParameters
        {
            Location = "<your-location>",
            Kind = DatabaseAccountKind.GlobalDocumentDB,
            ConsistencyPolicy = new ConsistencyPolicy
            {
                DefaultConsistencyLevel = DefaultConsistencyLevel.Session
            }
        };

        await client.DatabaseAccounts.StartCreateOrUpdateAsync(resourceGroupName, accountName, account);

        Console.WriteLine("Cosmos DB account provisioned successfully!");

        // List all Cosmos DB accounts in the resource group
        var accounts = client.DatabaseAccounts.ListByResourceGroup(resourceGroupName);

        Console.WriteLine("List of Cosmos DB accounts in the resource group:");

        await foreach (var dbAccount in accounts)
        {
            Console.WriteLine(dbAccount.Name);
        }

        // Update the Cosmos DB account
        var updateAccount = new DatabaseAccountUpdateParameters
        {
            Tags = { { "Environment", "Production" } }
        };

        await client.DatabaseAccounts.StartUpdateAsync(resourceGroupName, accountName, updateAccount);

        Console.WriteLine("Cosmos DB account updated successfully!");

        // Delete the Cosmos DB account
        await client.DatabaseAccounts.StartDeleteAsync(resourceGroupName, accountName);

        Console.WriteLine("Cosmos DB account deleted successfully!");
    }
}